File: grind-75/array/majority_element.py
from typing import List


class Solution:
    @classmethod
    def majorityElement(self, nums: List[int]) -> int:
        """
        :type nums: List[int]
        :rtype: int
        """
        for num in set(nums):
            if nums.count(num) > len(nums) // 2:
                return num

File: projects/grind-75/typescript/array/majority-element.ts
export default function majorityElement(nums: number[]): number {
  let candidate: number | undefined;
  let count = 0;
  for (let num of nums) {
    if (count == 0) {
      candidate = num;
    }
    count += num === candidate ? 1 : -1;
  }
  return candidate!;
}
